common/ayon_common/distribution/control.py

```diff
@@ -500,33 +744,69 @@ def _distribute(self):
             self.state = UpdateState.MISS_SOURCE_FILES
             return
 
+        # Progress file
+        if self._progress_dir:
+            # Check if other process/machine already started the job
+            if _wait_for_other_process(
+                self._progress_dir,
+                self._progress_id,
+                self.log
+            ):
+                self.state = UpdateState.UPDATED
+                self.log.info(f"{self.item_label}: Already distributed")
+                return
```

Comment:
It could be merged together

```suggestion
        if self._progress_dir and _wait_for_other_process(
                        self._progress_dir,
                        self._progress_id,
                        self.log
                    ):
            self.state = UpdateState.UPDATED
            self.log.info(f"{self.item_label}: Already distributed")
            return
```